colander/__init__.py

```diff
@@ -1887,6 +1887,8 @@ def serialize(self, appstruct=null):
         if isinstance(appstruct, deferred): # unbound schema with deferreds
             appstruct = null
         cstruct = self.typ.serialize(self, appstruct)
+        if self.missing is drop and cstruct is null:
+            cstuct = drop
```

Comment:
`cstuct` is a typo. As was said earlier this PR needs a test (which would have caught the bug in this PR).